src/expr/src/interpret.rs

```diff
@@ -298,16 +298,42 @@ impl<'a> ResultSpec<'a> {
             // Otherwise, if our function is monotonic, we can try mapping the input
             // range to an output range.
             Values::Within(min, max) if is_monotone => {
-                let min_column = result_map(Ok(min));
-                let max_column = result_map(Ok(max));
-                if min_column.nullable
-                    || min_column.fallible
-                    || max_column.nullable
-                    || max_column.fallible
-                {
-                    ResultSpec::anything()
-                } else {
-                    min_column.union(max_column)
+                let a_result = result_map(Ok(min));
+                let b_result = result_map(Ok(max));
+
+                match (a_result, b_result) {
+                    // If both endpoints give us a range, the result is a union of those ranges.
```

Comment:
```suggestion
                let min_result = result_map(Ok(min));
                let max_result = result_map(Ok(max));

                match (min_result, max_result) {
```

Reply:
Applied the rename but kept the comment. (I assumed removing the comment was just an accident, but if that's wrong let me know!)